nxontology_data/efo/queries/mapping_properties.rq

```diff
@@ -0,0 +1,28 @@
+PREFIX mondo: <http://purl.obolibrary.org/obo/mondo#>
+PREFIX efo: <http://www.ebi.ac.uk/efo/>
+PREFIX skos: <http://www.w3.org/2004/02/skos/core#>
+
+SELECT ?efo_id ?xref_id  ?mapping_property_id ?efo_uri ?xref_uri ?mapping_property_uri
+WHERE {
+  VALUES ?mapping_property_uri {mondo:closeMatch mondo:exactMatch skos:mappingRelation skos:closeMatch skos:exactMatch skos:broadMatch skos:narrowMatch skos:relatedMatch}
+
+  ?efo_uri rdf:type owl:Class .
+  ?efo_uri ?mapping_property_uri ?xref_uri
+
+
+  BIND( REPLACE( STR(?efo_uri), "^http.+/([^:]+)_(.+)$", "$1:$2" ) AS ?efo_id )
+  BIND( REPLACE( STR(?xref_uri), "^http.+/([^:]+)_(.+)$", "$1:$2" ) AS ?xref_id_dirty )
+  BIND(
+    COALESCE(
+        IF( STRSTARTS( ?xref_id_dirty, "http://identifiers.org" ), REPLACE( ?xref_id_dirty, "^http.*/(.+)/(.+)$", "$1:$2" ), ?error ),
+        IF( STRSTARTS( ?xref_id_dirty, "http://linkedlifedata.com/resource/umls/id" ), REPLACE( ?xref_id_dirty, "^http.*/(.+)$", "UMLS:$1" ), ?error ),
+        IF( STRSTARTS( ?xref_id_dirty, "http://purl.bioontology.org/ontology/ICD10CM" ), REPLACE( ?xref_id_dirty, "^http.*/(.+)$", "ICD10CM:$1" ), ?error ),
+        IF( STRSTARTS( ?xref_id_dirty, "https://icd.who.int/browse10/2019/en#" ), REPLACE( ?xref_id_dirty, "^http.*/(.+)$", "ICD10:$1" ), ?error ),
+        IF( STRSTARTS( ?xref_id_dirty, "https://omim.org/entry" ), REPLACE( ?xref_id_dirty, "^http.*/(.+)$", "OMIM:$1" ), ?error ),
+        IF( STRSTARTS( ?xref_id_dirty, "https://omim.org/phenotypicSeries" ), REPLACE( ?xref_id_dirty, "^http.*/PS(.+)$", "OMIMPS:$1" ), ?error ),
```

Comment:
All these special cases are a bit annoying to maintain, but great work figuring them out. Was it just an iterative process of figuring out which URIs are not handled?

One option would be to save the URI to CURIE conversion for post-processing in python with [`bioregistry.curie_from_iri`](https://bioregistry.readthedocs.io/en/latest/api/bioregistry.curie_from_iri.html#bioregistry.curie_from_iri) or [`curies`](https://github.com/cthoyt/curies).

Reply:
> Was it just an iterative process of figuring out which URIs are not handled?

Yes, this was an iterative process



> One option would be to save the URI to CURIE conversion for post-processing in python with [`bioregistry.curie_from_iri`](https://bioregistry.readthedocs.io/en/latest/api/bioregistry.curie_from_iri.html#bioregistry.curie_from_iri) or [`curies`](https://github.com/cthoyt/curies).

Thanks for the tip, this would be better if we can do it like that.  

I used `curies.get_bioregistry_converter` to get the converter for URIs. However, there are some differences how `curies` maps URIs compared to how I mapped them.  Here are some examples:
- `orphanet:99022` vs `obo:orphanet_99022` URI: `http://purl.obolibrary.org/obo/Orphanet_99022`.  Should we replace `obo:orphanet_` with `orphanet:`?
-  `orphanet:98813` vs `orphanet.ordo:98813` URI: `http://www.orpha.net/ORDO/Orphanet_98813` I guess we can replace `orphanet.ordo` with `orphanet` akin to https://github.com/related-sciences/nxontology-data/blob/c7b14299b18ed391762758bf76d8f5f806006a60/nxontology_data/utils.py#L84-L88
- `omimps:203655` vs `omim.ps:203655` URI: `https://omim.org/phenotypicSeries/PS203655` . Should we replace `omim.ps` with `omimps`?


There is also a missing uri_prefix `http://purl.bioontology.org/ontology/ICD10CM/` for `ICD10CM`.  The [add_prefix](https://curies.readthedocs.io/en/latest/api/curies.Converter.html#curies.Converter.add_prefix) method for adding prefixes lacks `merge` option in the version we use. Would it be safe to update the `curies` version?

Reply:
> * Should we replace `obo:orphanet_` with `orphanet:`?

Yes. It's nice that curies handles that.

For orphanet, we can replace `orphanet.ordo` after normalization.

`omim.ps` is the correct normalized prefix.

Ideal is you call `normalize_parsed_curie` on the output of `curies.get_bioregistry_converter` so we get consistently formatted CURIES everywhere.